src/polymer/element-descriptor.ts

```diff
@@ -182,33 +182,36 @@ function resolveElement(
 }
 
 function getFlattenedAndResolvedBehaviors(
-    behaviors: string[], document: Document) {
+    behaviors: ScannedBehaviorAssignment[], document: Document) {
   const resolvedBehaviors = new Set<Behavior>();
   _getFlattenedAndResolvedBehaviors(behaviors, document, resolvedBehaviors);
   return resolvedBehaviors;
 }
 
 function _getFlattenedAndResolvedBehaviors(
-    behaviors: string[], document: Document, resolvedBehaviors: Set<Behavior>) {
-  const toLookup = behaviors.slice();
-  for (let behaviorName of toLookup) {
+    behaviors: ScannedBehaviorAssignment[], document: Document,
```

Comment:
this was in the existing code, but rereading it I think that the variable name `behaviors` might be better as something like `behaviorReferences`